### PR TITLE
Update BO architecture in product settings page

### LINKED ISSUE #24769: Reorganize fields in product settings page
#### Is your feature request related to a problem?

For a better architecture, reorganize fields in BO > shop parameters > product settings page in the following order:

product settings

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Reorganize fields n BO > shop parameters > product settings page for a better architecture
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | yes / no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24769.
| How to test?      | BO > shop parameters > product settings verify if fields are saved correctly 
| Possible impacts? | Product pages in FO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27440)

### CODE REVIEW

**Comment by nesrineabdmouleh** [comment]
hello @PierreRambaud, @kevin-xtn
When updating UI tests, I found that the last label is missing on Product Settings block
=> See attached screenshot

screenshot

Thank you
Nesrine

**Review by jolelievre — APPROVED** [review]
Thanks @PierreRambaud

**Review by jolelievre — COMMENTED** [review]
Oh actually there is a feedback from QA tests

**Review by jolelievre — DISMISSED** [review]
Seems like a label is missing

**Review by atomiix — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by florine2623 — DISMISSED** [review]
Hello @PierreRambaud ,

I have some differences between the [mockup](https://github.com/PrestaShop/PrestaShop/issues/24769#issue-909577901) and the result with PR.

What I have :
Screenshot 2022-03-07 at 16 23 46
What I should have : 
Screenshot 2022-03-07 at 16 23 51

Here are some changes to add in Product Stock block :
Screenshot 2022-03-07 at 17 46 40


Could you check please ?
Thanks!

**Comment by PierreRambaud** [comment]
> Hello @PierreRambaud ,
> 
> I have some differences between the [mockup](https://github.com/PrestaShop/PrestaShop/issues/24769#issue-909577901) and the result with PR.
> 
> What I have : Screenshot 2022-03-07 at 16 23 46 What I should have : Screenshot 2022-03-07 at 16 23 51
> 
> Here are some changes to add in Product Stock block : Screenshot 2022-03-07 at 17 46 40
> 
> Could you check please ? Thanks!

It's a first improvement.
I can't change all wordings as wordings has been accepted by @Julievrz and if we change them, we have to change a lot  of things.
I also can't remove the field at the bottom as it's a new field that has been added after mockup

**Comment by Julievrz** [comment]
The thing is that, on the mockups, it's written "Stock management" because the toggle button indicates "Enabled/Disabled" instead of "Yes/No". It makes sense. But if the toggle button is "Yes/No" (and I think it can't be changed to "Enabled/Disabled" that easily*), then it's clearer to start with a verb "Enable stock management" (?) "Yes/No".

* An issue was opened about this: https://github.com/PrestaShop/PrestaShop/issues/19142

I agree with @PierreRambaud that wording improvements could be done later if the wording of toggle buttons is modified everywhere in the software (linked to the issue I've just mentioned).

File: src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/Blocks/product_preferences_stock.html.twig [hunk 1]
```diff
@@ -51,6 +43,33 @@
...
+          <div class="col-sm">
+            {{ form_errors(stockForm.display_unavailable_attributes) }}
+            {{ form_widget(stockForm.display_unavailable_attributes) }}
+          </div>
+        </div>
+
+        {{ form_row(stockForm.display_last_quantities) }}
+
+        <div class="form-group row">
+          {{ ps.label_with_help(('Order out-of-stock products'|trans), ('By default, the "%add_to_cart_label%" button is hidden when a product is unavailable. You can choose to have it displayed in all cases.'|trans({'%add_to_cart_label%': 'Add to cart'|trans({}, 'Shop.Theme.Actions')}, 'Admin.Shopparameters.Help'))) }}
```
Comment (Julievrz) [inline src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/Blocks/product_preferences_stock.html.twig]: Let's stick with the current wording. We'll change it later if necessary when the toggle buttons are modified. Thanks!

```suggestion
          {{ ps.label_with_help(('Allow ordering of out-of-stock products'|trans), ('By default, the "%add_to_cart_label%" button is hidden when a product is unavailable. You can choose to have it displayed in all cases.'|trans({'%add_to_cart_label%': 'Add to cart'|trans({}, 'Shop.Theme.Actions')}, 'Admin.Shopparameters.Help'))) }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/Blocks/product_preferences_general.html.twig [hunk 2]
```diff
@@ -31,7 +31,7 @@
   <div class="card" id="configuration_fieldset_products">
     <h3 class="card-header">
       <i class="material-icons">settings</i>
-      {{ 'Products (general)'|trans }}
+      {{ 'Products general settings'|trans }}
```
Comment (Julievrz) [inline src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/Blocks/product_preferences_general.html.twig]:
```suggestion
      {{ 'General product settings'|trans }}
```

Sorry I forgot to modify this one

**Review by Progi1984 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @PierreRambaud ,

As seen with @MatShir , the wording is to be ignored.
The position of fields in all 4 blocks is correct. Editing and Saving the configuration in all blocks work well.

It is QA ✅ 
Thanks @PierreRambaud :)

**Comment by Progi1984** [comment]
Thanks @PierreRambaud & @florine2623

**Pull request merged**